File: src/main/java/org/jdc/template/Analytics.java
package org.jdc.template;

import com.google.android.gms.analytics.Tracker;

import java.util.Map;

public interface Analytics {
    // Categories
    String CATEGORY_APP = "App";
    String CATEGORY_INDIVIDUAL = "Individual";
    String CATEGORY_ABOUT = "About";
    String CATEGORY_SETTINGS = "Settings";

    // Actions
    String ACTION_APP_LAUNCH = "Launch";
    String ACTION_NEW = "New";
    String ACTION_VIEW = "View";
    String ACTION_EDIT = "Edit";
    String ACTION_EDIT_SAVE = "Edit Save";
    String ACTION_DELETE = "Delete";

    // Variables
    String VARIABLE_BUILD_TYPE = "Build Type";

    void send(Map<String, String> params);

    class GoogleAnalytics implements Analytics {
        private final Tracker tracker;

        public GoogleAnalytics(Tracker tracker) {
            this.tracker = tracker;
        }

        @Override
        public void send(Map<String, String> params) {
            tracker.send(params);
        }
    }
}

File: src/main/kotlin/org/jdc/template/Analytics.kt
package org.jdc.template

import com.google.android.gms.analytics.Tracker

interface Analytics {

    fun send(params: Map<String, String>)

    class GoogleAnalytics(private val tracker: Tracker) : Analytics {

        override fun send(params: Map<String, String>) {
            tracker.send(params)
        }
    }

    companion object {
        // Categories
        val CATEGORY_APP = "App"
        val CATEGORY_INDIVIDUAL = "Individual"
        val CATEGORY_ABOUT = "About"
        val CATEGORY_SETTINGS = "Settings"

        // Actions
        val ACTION_APP_LAUNCH = "Launch"
        val ACTION_NEW = "New"
        val ACTION_VIEW = "View"
        val ACTION_EDIT = "Edit"
        val ACTION_DELETE = "Delete"

        // Variables
        val VARIABLE_BUILD_TYPE = "Build Type"
    }
}